pom.xml

```diff
@@ -96,6 +96,22 @@
   </build>
 
   <profiles>
+    <profile>
+      <id>non-essential</id>
```

Comment:
Can you please explain what purpose this serves exactly?
Since the Python stuff apparently doesn't re-build anything, I don't understand why this profile needs to exist.
Plus I really don't like having entire modules marked as "non essential".

Reply:
It does rebuild it, and doesn't require  the quarkus/spring/other stuff, which would only serve to pollute the Python source distribution.

Reply:
Maybe it doesn't need to rebuild it anymore? It's the same repo, let's start treating it like that. Python will not be built separately, as an extra thing. They will be built together.

Reply:
This allows :

- `python -m build` / `pip install` workflows will ALWAYS build latest sources without requiring a previous `mvn clean install`

Previously, this wasn't possible, because the sources lived in different repositories.

Reply:
(or put another way, if you change a Java file, you can just run `tox` without needing to remember to do `mvn clean install` before)

Reply:
I understand, but... why would I even _want_ a rebuild if I've just built it?
`mvn clean install` in the top level, that needs to be the goal, for our own dev experience.

Reply:
If you are working on Python, you never need to do `mvn clean install` or any familiarity of Java build tools at all.
This just makes all the tools in the Python ecosystem (`pip install`, `tox`, `build`, ...) just work with the Java code, without requiring any work.
This means if a user wanted to build the latest sources, this will work even if they don't have `mvn` installed and haven't built Timefold Solver locally:

```bash
pip install git+https://github.com/TimefoldAI/timefold-solver.git
```
 
And how are we suppose to determine if it is just built and not a stale copy?

Reply:
> If you are working on Python, you never need to do `mvn clean install` or any familiarity of Java build tools at all.

There is no "working on Python". Every feature we had since the Python solver was introduced ended up requiring Python code as well. I'm not ruling out some changes will be exclusive to Python, but there will be a relatively small amount of them. IMO the development workflow needs to be tailored for what happens most often, not for corner cases. And from my current experience, the dominant workflow is working on Java _and_ Python.

> And how are we suppose to determine if it is just built and not a stale copy?

We aren't. Or rather the build system isn't supposed to. It's the developer's responsibility - just as it is currently with Java. If I run tests on stale Java code, I'll run into issues.

Reply:
That what I mean! You don't need to run `mvn clean install` to run Python tests! How are you running Python tests. This simplifies things, not complicates it.

Reply:
I could argue it simplifies Python things while complicating the Java POM. 

But I see your point. Let's come up with a better name than `non-essential` and I'll shut up. Perhaps something that describes purpose of the profile.

Reply:
`java-tools-and-integrations`?